x-pack/legacy/plugins/maps/public/layers/vector_layer.js

```diff
@@ -585,6 +596,34 @@ export class VectorLayer extends AbstractLayer {
     });
   }
 
+  _setMbTextProperties(mbMap) {
+    const sourceId = this.getId();
+    const textLayerId = this._getMbTextLayerId();
+    const textLayer = mbMap.getLayer(textLayerId);
+
+    if (!textLayer) {
+      mbMap.addLayer({
+        id: textLayerId,
+        type: 'symbol',
+        source: sourceId,
+      });
+    }
+
+    const filterExpr = getPointFilterExpression(this._hasJoins());
+    if (filterExpr !== mbMap.getFilter(textLayerId)) {
```

Comment:
in hindsight, this reference-equality check is imho not best practice. it relies on mb keeping the reference to the filter, which isn't documented.

Consider using `_.isEqual`, and then change in the other mbSync methods too

Reply:
I created https://github.com/elastic/kibana/issues/53246 to track.